node/lib/cmd/commit.js

```diff
@@ -105,12 +112,49 @@ the URL of submodules).`,
     });
 };
 
+const execPreCommitHooks = co.wrap(function *(repo) {
+    const Hook          = require("../util/hook");
+    const SubmoduleUtil = require("../util/submodule_util");
+
+    // execute pre-commit hook for each submodule
+    const submoduleNames = yield SubmoduleUtil.getSubmoduleNames(repo);
+    for (const submoduleName of submoduleNames) {
+        const subRepo = yield SubmoduleUtil.getRepo(repo, submoduleName);
+        const status = yield Hook.execHook(subRepo, "pre-commit");
+        if (status === 1) {
+            console.error(
+              "Failed to commit. \n" +
+              "Error occurred when running pre-commit hook of submodule: " +
+              submoduleName
+            );
+            process.exit(1);
+        }
+    }
+});
+
 const doCommit = co.wrap(function *(args) {
     const Commit  = require("../util/commit");
     const GitUtil = require("../util/git_util");
     const Hook    = require("../util/hook");
 
     const repo = yield GitUtil.getCurrentRepo();
+
+    if (!args.no_verify) {
+        const usingPaths = 0 !== args.file.length;
+        const usingInteractive = args.interactive;
+        if (usingPaths) {
+            console.warn(
```

Comment:
1. I would really like to handle this case, if at all possible
2. I don't want to warn someone that precommit hooks are skipped if they don't have any hooks.

Reply:
I just pushed a new commit that should address both of your points.
* It should handle use [files]
* It should only warn users if there is any hooks that need to be run (submodule has 'pre-commit' hook && submodule has changes to commit) but skipped.